gnomad_qc/v4/variant_qc/evaluation.py

```diff
@@ -193,55 +269,44 @@ def main(args):  # noqa: D103
         )
 
     if args.extract_truth_samples:
-        logger.info(f"Extracting truth samples from MT...")
-        mt = get_gnomad_v3_mt(
-            key_by_locus_and_alleles=True, remove_hard_filtered_samples=False
-        )
-
-        mt = mt.filter_cols(
-            hl.literal([v["s"] for k, v in TRUTH_SAMPLES.items()]).contains(mt.s)
-        )
-        mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
-
-        # Checkpoint to prevent needing to go through the large table a second time
-        mt = mt.checkpoint(
-            get_checkpoint_path("truth_samples", mt=True),
-            overwrite=args.overwrite,
+        logger.info(f"Extracting truth samples from VDS...")
+        res = evaluation_resources.extract_truth_samples
+        res.check_resource_existence()
+        vds = get_gnomad_v4_vds(
+            controls_only=True, split=True, filter_partitions=range(2) if test else None
         )
-
-        for truth_sample in TRUTH_SAMPLES:
-            truth_sample_mt = mt.filter_cols(mt.s == TRUTH_SAMPLES[truth_sample]["s"])
-            # Filter to variants in truth data
-            truth_sample_mt = truth_sample_mt.filter_rows(
-                hl.agg.any(truth_sample_mt.GT.is_non_ref())
-            )
-            truth_sample_mt.naive_coalesce(args.n_partitions).write(
-                get_callset_truth_data(truth_sample).path,
-                overwrite=args.overwrite,
-            )
+        # Checkpoint to prevent needing to go through the large VDS multiple times.
+        vds = vds.checkpoint(new_temp_file("truth_samples", "vds"), overwrite=True)
```

Comment:
do you need overwrite=True if you're using new_temp_file ? I thought it was a new name each time so an overwrite wasn't needed

Reply:
yeah, I guess you don't really, I can remove